src/ThreadsCommon.c

```diff
@@ -0,0 +1,160 @@
+		if (!cyclic_reqeust_data[index].is_used &&
+		    cyclic_reqeust_data[index].timer_id == timer_id) {
+			break;
+		}
+	}
+
+	cyclic_reqeust_data[index].next_wakeup_ticks +=
+	    cyclic_reqeust_data[index].interval_ticks;
+	rtems_interval delta = cyclic_reqeust_data[index].next_wakeup_ticks -
+			       rtems_clock_get_ticks_since_boot();
+	rtems_timer_fire_after(timer_id, delta, timer_callback, NULL);
+}
+
+static void timer_callback(rtems_id timer_id, void *unused)
+{
+	uint32_t index = 0;
+	for (uint32_t index = 0; index < RT_MAX_CYCLIC_INTERFACES; index++) {
+		if (!cyclic_reqeust_data[index].is_used &&
+		    cyclic_reqeust_data[index].timer_id == timer_id) {
+			break;
+		}
+	}
+
+	rtems_message_queue_send(cyclic_reqeust_data[index].queue_id,
+				 &empty_request,
+				 cyclic_reqeust_data[index].request_size);
+	schedule_next_tick(cyclic_reqeust_data[index].timer_id);
+}
+
+static void update_execution_time_data(const uint32_t thread_id,
+				       const uint64_t thread_execution_time)
+{
+
+	if (thread_execution_time <
+	    threads_info[thread_id].min_thread_execution_time) {
+		threads_info[thread_id].min_thread_execution_time =
+		    thread_execution_time;
+	}
+
+	if (thread_execution_time >
+	    threads_info[thread_id].max_thread_execution_time) {
+		threads_info[thread_id].max_thread_execution_time =
+		    thread_execution_time;
+	}
+
+	threads_info[thread_id].mean_thread_execution_time =
+	    threads_info[thread_id].mean_thread_execution_time +
+	    ((double)thread_execution_time -
+	     threads_info[thread_id].mean_thread_execution_time) /
+		((double)threads_info[thread_id].execution_time_counter + 1);
+
+	threads_info[thread_id].execution_time_counter++;
+}
+
+bool ThreadsCommon_CreateCyclicRequest(uint64_t interval_ns, uint32_t queue_id,
+				       uint32_t request_size)
+{
+	bool empty_slot_found = false;
+	uint32_t index = 0;
+
+	for (index = 0; index < RT_MAX_CYCLIC_INTERFACES; index++) {
+		if (!cyclic_reqeust_data[index].is_used) {
+			empty_slot_found = true;
+		}
+	}
+
+	if (!empty_slot_found) {
+		return false;
+	}
+
+	rtems_name name = generate_new_partition_timer_name();
+	const rtems_status_code timer_creation_status =
+	    rtems_timer_create(name, &cyclic_reqeust_data[index].timer_id);
+	if (timer_creation_status != RTEMS_SUCCESSFUL) {
+		return false;
+	}
+
+	cyclic_reqeust_data[index].next_wakeup_ticks = 0;
+	cyclic_reqeust_data[index].interval_ticks =
+	    RTEMS_MILLISECONDS_TO_TICKS(interval_ns * NANOSECOND_IN_MILISECOND);
+	cyclic_reqeust_data[index].queue_id = queue_id;
+	cyclic_reqeust_data[index].request_size = request_size;
+	cyclic_reqeust_data[index].is_used = true;
+
+	schedule_next_tick(cyclic_reqeust_data[index].timer_id);
+
+	return true;
+}
+
+bool ThreadsCommon_ProcessRequest(void *request_data, uint32_t request_size,
+				  void *user_function, uint32_t thread_id)
+{
+	call_function casted_user_function = user_function;
```

Comment:
Cast is the proper past tense. And where is the cast? The compiler still does not complain?